### DISCUSSION TITLE
Multi-Site Active-Active Replication - not replicating large files

### DISCUSSION THREAD

**Discussion by effectivelywild** [minio/minio#16359] [Q&A] [answered]
I am in the early stages of testing minio and have configured three clusters, one at each site using 4 nodes and 4 discs each - they are currently not behind any proxies. 

I have setup Multi-Site Active-Active Replication successfully and all statuses look healthy in the minio console and using _mc admin replicate_.

I have replicated thousands of objects across all sites without issue but there appears to be issues replicating larger files. I don't have an exact size, but it appears to be anything over 1.2GB.

When I use _mc replicate status_ I see a Size (Bytes) field that indicates the amount of data that has failed to replicate for each site. This is the exact total of all the larger files I have placed in the bucket to be replicated. 

I am unsure how I should debug this issue. 

I do see one "failed to get lock for object" error in the logs, and it is one of the larger files to be replicated. But I only see this once, for one file, there are dozens.

When using mc to trace logs I see multiple 409 lock conflicts from all clusters but there isn't any information to go on, or if this is even related to replication.

**Comment by harshavardhana** [comment, marked as answer]
What is your MinIO version here ?

**Reply by effectivelywild** [reply]
minio version RELEASE.2022-11-26T22-43-32Z (commit-id=63fc6ba2cd174ec2cbac2b022314c97bad6b6e4d)
Runtime: go1.19.3 linux/amd64
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Copyright: 2015-2022 MinIO, Inc.

**Reply by harshavardhana** [reply]
Please upgrade to latest and redo your tests.

**Reply by klauspost** [reply]
Also lock conflicts are expected. That is the nature of locks.

**Reply by ravindk89** [reply]
@fmuise also just to check - are there any load balancers, reverse proxies, or similar network control components sitting in front of your clusters? anything that might add timeouts, req/res body size limits, things of that nature?

**Reply by effectivelywild** [reply]
> @fmuise also just to check - are there any load balancers, reverse proxies, or similar network control components sitting in front of your clusters? anything that might add timeouts, req/res body size limits, things of that nature?

Thanks - there's nothing in front of them right now, I wanted to do some basic testing before the proxies and load balancers were added. Any idea where I could verify if there are timeouts with specific objects? Could it be caused it latency between sites? Are there site connectivity bandwidth/latency requirements?

**Reply by effectivelywild** [reply]
> Please upgrade to latest and redo your tests.

I will redelpoy the cluster today and test.

**Reply by harshavardhana** [reply]
> Thanks - there's nothing in front of them right now, I wanted to do some basic testing before the proxies and load balancers were added. Any idea where I could verify if there are timeouts with specific objects? Could it be caused it latency between sites? Are there site connectivity bandwidth/latency requirements?

Mostly your replication network is saturated, you have to size you cluster accordingly.

To get sizing and architecture guidelines you should get subscription with us.

Building large clusters in silo is not worth your time, you should get access to be our SUBNET to have SLA backed replies.

**Reply by effectivelywild** [reply]
Deployed the latest version this morning with no changes to my deployment logic/networking and there are no replication issues.

For your information I was running RELEASE.2022-11-26T22-43-32Z and moved to RELEASE.2023-01-02T09-40-09Z.

Thanks
